cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/optimizer/ExpressionContext.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.cohort.cql.spark.optimizer;
+
+import java.util.Stack;
+
+import org.hl7.elm.r1.ExpressionDef;
+import org.hl7.elm.r1.Query;
+import org.hl7.elm.r1.VersionedIdentifier;
+
+/**
+ * Maintains state for an expression definition so that aliases and let
+ * references can be resolved by other parts of the expression
+ */
+public class ExpressionContext {
+
+    private VersionedIdentifier libraryIdentifier;
+    private ExpressionDef expression;
+
+    public ExpressionContext(VersionedIdentifier libraryIdentifier, ExpressionDef elm) {
+        this.libraryIdentifier = libraryIdentifier;
+        this.expression = elm;
+    }
+    
+    public VersionedIdentifier getLibraryIdentifier() {
+        return libraryIdentifier;
+    }
+    public ExpressionDef getExpression() {
+        return expression;
+    }
+    
+    private Stack<QueryContext> queryContextStack = new Stack<QueryContext>();
```

Comment:
This was a result of copy/paste job from the OSS code. I typically try to avoid any synchronized collection when possible, but I didn't notice that this one creeped in. There is a [slight behavior difference](https://www.baeldung.com/java-deque-vs-stack) between Stack and Deque related to iteration, but I think Deque does it the way that you'd actually want anyway (top-down vs. bottom-up). I will change it.